### PR TITLE
mqtt: Improve handling of multiple PDU parsing

### PR DESCRIPTION
**Pull request by jlucovsky**
Issue: 6592

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: [6592](https://redmine.openinfosecfoundation.org/issues/6592)

Describe changes:
- Parse PDU instead of entire stream

### Provide values to any of the below to override the defaults.

To use a pull request use a branch name like `pr/N` where `N` is the
pull request number.

Alternatively, `SV_BRANCH` may also be a link to an
OISF/suricata-verify pull-request.

```
SV_REPO=
SV_BRANCH=
SU_REPO=
SU_BRANCH=
LIBHTP_REPO=
LIBHTP_BRANCH=
```

### CODE REVIEW

**Comment by catenacyber** [comment]
Could you have a SV test ? Maybe adding `bsize` To the current mqtt-frames test ?

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 17328

File: rust/src/mqtt/mqtt.rs [hunk 1]
```diff
@@ -553,8 +553,8 @@ impl MQTTState {
                     let _pdu = Frame::new(
                         flow,
                         &stream_slice,
-                        input,
-                        input.len() as i64,
+                        current,
+                        current.len() as i64,
```
Comment (catenacyber) [inline rust/src/mqtt/mqtt.rs:557]: This should not be `current.len()` as current may hold multiple PDUs, but rather `start.len() - rem.len()`

**Review by catenacyber — CHANGES_REQUESTED** [review]
CI needs to be green as well

**Pull request closed without merging**